extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java

```diff
@@ -408,6 +408,7 @@ private void handleNewWorkOrder(
 
     final QueryContext queryContext = task != null ? QueryContext.of(task.getContext()) : QueryContext.empty();
     final RunWorkOrder runWorkOrder = new RunWorkOrder(
+        task.getControllerTaskId(),
```

Comment:
We have UT's for this no ? I think we should have caught this as part of refactor ?

Reply:
The stage id and controller task id seem to be always the same for now for all queries after the controller refactor, so it would not get caught in one branch. I think this would only be caught during an upgrade scenario where IDs were getting randomly generated from previous versions of controller, or we add some functionality which causes a worker id to have other stage ids.